src/Symfony/Bundle/TwigBundle/TemplateIterator.php

```diff
@@ -22,21 +22,18 @@
 class TemplateIterator implements \IteratorAggregate
 {
     private $kernel;
-    private $rootDir;
     private $templates;
     private $paths;
     private $defaultPath;
 
     /**
      * @param KernelInterface $kernel      A KernelInterface instance
-     * @param string          $rootDir     The directory where global templates can be stored
      * @param array           $paths       Additional Twig paths to warm
      * @param string          $defaultPath The directory where global templates can be stored
      */
-    public function __construct(KernelInterface $kernel, string $rootDir, array $paths = [], string $defaultPath = null)
+    public function __construct(KernelInterface $kernel, array $paths = [], string $defaultPath)
```

Comment:
In theory it's missing a deprecation in 4.4. But there are several compiler passes that expect the paths at a certain argument number. So either we update the compiler passes like done in this PR but that might not work when people overwrote the definition. Or we don't update the compiler passes which would trigger internal deprecations.
The only solution I see is to mark this class as `@internal`. To me this class is an implementation detail and should not be relied on as an extension point. This I guess is also why `TemplateCacheWarmer` does not typhint this class but `iterable`.

Reply:
#31828